src/domain/core/src/entities/metadata_chain.rs

```diff
@@ -238,6 +232,64 @@ pub trait MetadataChainExt: MetadataChain {
 
         Ok(())
     }
+
+    /// An auxiliary method that simplifies the work if only one Visitor is
+    /// used.
+    async fn accept_one<V, E>(&self, mut visitor: V) -> Result<V, E>
+    where
+        V: MetadataChainVisitor<Error = E>,
+        E: Error + From<IterBlocksError>,
+    {
+        self.accept(&mut [&mut visitor]).await?;
+
+        Ok(visitor)
+    }
+
+    /// An auxiliary method that simplifies the work if only one custom Visitor
+    /// is used.
+    async fn reduce<S, F, E>(
+        &self,
+        state: S,
+        initial_decision: MetadataVisitorDecision,
+        callback: F,
+    ) -> Result<S, E>
+    where
+        S: Send + Sync,
```

Comment:
> I suspect that this stems from MetadataChainVisitor being Send + Sync

Yes, you're right, that's the reason

> I don't see a case where we would access one visitor from multiple threads

in this case, the asynchrony is contributing to it -- since, I used mutable refs to a slice with Visitors, a future is not considered appropriate if the ref type inside the slice is not `Send`

We have several options:
1) we can leave it as is -- that's what I'd like to suggest
2) remove links and replace with owning a list of Visitors in the heap (`Vec`)
3) remove `Send` from `MetadataChainVisitor` и box futures that are waiting for `Send`

Reply:
`Sync` bound has been removed -- Done